com.unity.perception/Runtime/Randomization/Samplers/SamplerTypes/AnimationCurveSampler.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+using Unity.Burst;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+using Unity.Jobs;
+using UnityEngine.Experimental.Perception.Randomization.Scenarios;
+
+namespace UnityEngine.Experimental.Perception.Randomization.Samplers
+{
+    /// <summary>
+    /// Returns random values according to a range and probability distribution denoted by a user provided AnimationCurve. The X axis of the AnimationCurve corresponds to the values this sampler will pick from, and the Y axis corresponds to the relative probability of the values. The relative probabilities (Y axis) do not need to max out at 1, as only the shape of the curve matters. The Y values cannot however be negative.
+    /// </summary>
+    [Serializable]
+    public class AnimationCurveSampler : ISampler
+    {
+        /// <summary>
+        /// The range field is not used by the AnimationCurve sampler.
+        /// </summary>
+        [field: NonSerialized]
+        public FloatRange range { get; set; }
+
+        /// <summary>
+        /// The Animation Curve associated with this sampler
+        /// </summary>
+        [Tooltip("Probability distribution curve used for this sampler. The X axis corresponds to the values this sampler will pick from, and the Y axis corresponds to the relative probability of the values. The relative probabilities (Y axis) do not need to max out at 1 as only the shape of the curve matters. The Y values cannot however be negative.")]
+        public AnimationCurve distributionCurve;
+
+        float[] m_IntegratedCurve;
+        bool m_CurveValid;
+        float m_StartTime;
+        float m_EndTime;
+        float m_Interval;
+        int m_NumOfSamples = 100;
```

Comment:
If m_NumOfSamples is a constant we should mark it as such. If it should be configurable by a user, I would make this setting public so they can initialize it in an initializer during construction of the sampler.

Reply:
Good point. I made it public and added a tooltip to explain.